src/Bicep.Core/TypeSystem/Az/ResourceTypeProviderFactory.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Azure.Bicep.Types;
+using Azure.Bicep.Types.Az;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Features;
+using Bicep.Core.Modules;
+using Bicep.Core.Registry.Oci;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.TypeSystem.Az;
+using Newtonsoft.Json;
+
+namespace Bicep.Core.TypeSystem
+{
+    public class ResourceTypeProviderFactory : IResourceTypeProviderFactory
+    {
+
+        private readonly (string, string) BuiltInAzLoaderKey = ("az", IResourceTypeProvider.BuiltInVersion);
```

Comment:
Have you considered using `null` to represent built-in, rather than a magic string?

Reply:
I have but chosen against it, I think the value is to be used as a placeholder. I want to have a follow up PR that assigns the value of the NuGet version to `BuiltInVersion`. That way we dedupliate a scenario where someone has an explicit import whose version matches the builtin one.

I think this can be done in a separate PR though so that is why I am for now using a placeholder